framework/src/main/java/org/checkerframework/framework/flow/CFAbstractValue.java

```diff
@@ -480,16 +507,27 @@ private V upperBound(@Nullable V other, boolean shouldWiden) {
     TypeMirror lubTypeMirror =
         TypesUtils.leastUpperBound(
             this.getUnderlyingType(), other.getUnderlyingType(), processingEnv);
+    return upperBound(other, lubTypeMirror, shouldWiden);
+  }
 
+  /**
+   * Returns an upper bound of this and {@code other}.
+   *
+   * @param other an abstract value
+   * @param upperBoundTypeMirror the underlying type of the returned value
+   * @param shouldWiden true if the lub should perform widening
+   * @return the least upper bound of this and {@code other}
```

Comment:
Above it says "an upper bound", here it says "the least upper bound".  Please clarify.
Also, please write a comment discussing the relationship between this method and `leastUpperBound`.